newIDE/app/src/EventsSheet/index.js

```diff
@@ -1305,6 +1347,20 @@ export default class EventsSheet extends React.Component<Props, State> {
                             }}
                           />
                         )}
+                        {this.state.textEditedEvent && (
+                          <EventTextDialog
+                            event={this.state.textEditedEvent}
+                            onApply={() => {
+                              this.closeEventTextDialog();
+                              this._saveChangesToHistory();
+                            }}
+                            onClose={() =>
```

Comment:
`onClose={this.closeEventTextDialog}`

Reply:
Which you can read as English: "When [the user wants to] Close, then close the event text dialog" :)